src/wallet/wallet.cpp

```diff
@@ -2396,14 +2397,24 @@ bool CWallet::DelAddressBook(const CTxDestination& address)
             return false;
         }
         // Delete data rows associated with this address
-        batch.EraseAddressData(address);
+        if (!batch.EraseAddressData(address)) {
+            WalletLogPrintf("Error: cannot erase address book entry data\n");
+            return false;
+        }
+
+        // Delete purpose and name
+        if (!batch.ErasePurpose(dest) || !batch.EraseName(dest)) {
```

Comment:
In 2d770f6ae88d9ee6f1775e011f241274059fe275 "wallet: clean redundancies in DelAddressBook"

I'd prefer to have these occur separately so that a distinct error can be logged for each, that way we can discover which operation failed.